src/api.rs

```diff
@@ -502,16 +502,18 @@ impl Api {
     }
 
     /// Lists all the release file for the given `release`.
+    /// Optionally filtered by a list of checksums.
     pub fn list_release_files(
         &self,
         org: &str,
         project: Option<&str>,
         release: &str,
+        checksums: Option<&Vec<String>>,
```

Comment:
We can also make it `&Vec<String>` and pass `vec![]` around, but didn't want to make the unnecessary allocation 🤷

Reply:
Here's what I think:

You only call this function with checksums in one place, in `flag_uploaded_sources`. There you unconditionally allocate the vector of checksums anyway. I would suggest sorting the vector there (immediately after creation) and just passing `checksums: &[String]` (or `checksums: Option<&[String]>`, whichever you think makes the nicer API) here.